### ISSUE TITLE
model training plateaus

### ISSUE THREAD

**Issue by jyu-theartofml** [akanimax/pro_gan_pytorch#46]
In regards to the latent size, can you still get good results using a value of 256? I tried to train ProGan on an art dataset, and it seems to make progress until Depth 6. Please see attached pictures showing the output at different depths. I ran these using 64 epochs, default fade-in, and 64 for batching, wgan-gp for the loss function, and trained it on 8 GPUs. Thanks!

`Depth 2`
![image](https://user-images.githubusercontent.com/7095892/85208232-eeb32d00-b2eb-11ea-9789-281a544ec862.png)

`Depth 3`
![gen_3_54_70](https://user-images.githubusercontent.com/7095892/85208248-fecb0c80-b2eb-11ea-8bb0-14ebc08f3734.png)

`Depth 4`
![gen_4_54_70](https://user-images.githubusercontent.com/7095892/85208255-0b4f6500-b2ec-11ea-9f27-58d397c54ef5.png)

`Depth 5`
![gen_5_64_70](https://user-images.githubusercontent.com/7095892/85208259-186c5400-b2ec-11ea-8e94-55688b542bda.png)

`Depth 6`
![gen_6_64_70](https://user-images.githubusercontent.com/7095892/85208270-26ba7000-b2ec-11ea-9a52-73097512980f.png)

**Comment by jyu-theartofml** [comment]
@akanimax , when you train the ProGan models, how many GPUs do you usually use? I'm training on a data set of 9K, and using 8 GPUs and relativistic-hinge loss. For some reason, the quality of the generated images starts to deteriorate at layer 5.

**Comment by akanimax** [comment]
The last I tested this code was with 4 V100s 32 GB, and it still took about a week to finish on CelebA-HQ 1024x1024. It has 30K images. I think you can shorten the training if you have smaller dataset. Regarding the loss function, I'd recommend using WGAN-GP or Non-Saturating StandardGAN with R1 penalty. 

If you are planning on running the trainings now, I'd suggest you to use the code from this repository instead of the pip-package, which is outdated. I haven't gotten the time to update it :sweat_smile:. 

I am planning on training something for a session I'll be giving next week. Please feel free to shoot me an email if you'd like to share your dataset, and I can train it for you. I would only like to use it as a demo in my presentation, and I'll credit you for the images. 

Hope this helps :+1:

Cheers!
@akanimax
